src/ctim/schemas/openc2_network.clj

```diff
@@ -0,0 +1,104 @@
+(ns ctim.schemas.network_coa
+  (:require [schema.core :as s]
+            [ctim.schemas.openc2vocabularies :as openc2v]))
+
+
+
+
+(s/defschema BGPBlackhole
+  {:type (s/enum "BGPBlackhole")
+   :host s/Str})
+
+(s/defschema DNSSinkhole
+  {:type (s/enum "DNSSinkhole")
+   :host s/Str})
+
+(def Protocol
+  (s/enum "TCP"
+          "UDP"
+          "ICMP"
+          "Any"))
+
+(def ACLAction
+  (s/enum "ALERT" 
+          "DROP"
+          "DENY"
+          "LOG"
+          "PASS"
+          "REJECT"))
+
+(s/defschema Traffic
+  {:protocol Protocol
+   :sourceAddress s/Str
+   :sourcePort s/Str
+   :destinationAddress s/Str
+   :destinationPort s/Str})
+
+(s/defschema NetworkACL
+  {:type (s/enum "NetworkACL")
+   :traffic Traffic
+   :action ACLAction})
+
+
+(s/def vlanProfile
+  {:vlanTag s/Str})
+
+(s/def secGroupProfile
+  {:secGroupTag s/Str
+   :secGroupACL s/Str})
+
+
+(s/defschema Remediation
+  {:type (s/enum "Remediation")
+   :server s/Str
+   :containmentProfile (s/either vlanProfile
+                                 secGroupProfile)
+   :acl NetworkACL})
+
+(s/defschema NonSensitive
+  {:type (s/enum "NonSensitive")
+   :permissibleIps [s/Str]
+   :acl NetworkACL})
+
+(s/defschema Honeypot
+  {:type (s/enum "Honeypot")
+   :permissibleIps [s/Str]
+   :acl NetworkACL 
+   :routes {:prefix s/Str
+            :NextHop s/Str
+            :NextHopType s/Str}})
+
+(def Encapsulation
+  (s/enum "GRE"
+          "VXLAN"))
+
+(s/defschema BlockModifier
+  {:type (s/enum 
+           "Perimeter"
+           "Internal")
+   :method (s/conditional
+            #(= "NetworkACL" (:type %)) NetworkACL
+            #(= "BGPBlackhole" (:type %)) BGPBlackhole
+            #(= "DNSSinkhole" (:type %)) DNSSinkhole)})
+
+
+  
+
+(s/defschema ContainModifier
+  {:type (s/eq "Contain")
+   :method (s/conditional
```

Comment:
I still recoil at modifying schemas to work around incidental complexity in Swagger.  In OAI/OpenAPI-Specification#57 they acknowledge that they may fix this in a future release.  Lack of "oneOf" (eg "conditional") is, at best, a feature that swagger doesn't support yet, at worst, a bad engineering decision.

In response to Yann's bullets:,

- I don't see a ton of value in automatic data generation based on schemas alone since it ignores the semantics of your data and it gives a false sense of confidence that you are testing the right thing.  
- I don't believe that it makes mapping impossible.  You can create multiple tables/indexes, one for each type (there is a small set of possible schemas), and do a join based on ID and type.

We should not be complicating @jyoverma's life any further with this argument.  :-)  @craigbro came up with a good plan that we should follow:

> [11:37 AM] Craig Brozefsky: my evil plan would be

> [11:37 AM] Craig Brozefsky: we just have a dummied schema for swagger
> [11:37 AM] Craig Brozefsky: but our stored schema is fully conditionalized and all that
> [11:38 AM] Craig Brozefsky: so we get all the prismatic specificity, and then if we fix the tooling, we drop the dummy